### PR TITLE
Verify loader version compatibility

### PR DESCRIPTION
**Pull request by Pessimistress**
Follow up of https://github.com/uber-web/loaders.gl/pull/453

### CODE REVIEW

File: modules/core/src/lib/loader-utils/parse-with-worker.js [hunk 1]
```diff
@@ -31,8 +31,8 @@ export default function parseWithWorker(loader, data, options, context) {
 
   return workerFarm.process(workerSource, `loaders.gl-${workerName}`, {
     arraybuffer: toArrayBuffer(data),
-    options,
-    source: 'loaders.gl', // Lets worker ignore unrelated messages
+    options, // __VERSION__ is injected by babel-plugin-version-inline
+    /* global __VERSION__ */ source: `loaders.gl@${__VERSION__}`, // Lets worker ignore unrelated messages
```
Comment (ibgreen) [inline modules/core/src/lib/loader-utils/parse-with-worker.js]: Nit: global at top, like other files...

File: modules/core/src/utils/version.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import assert from './assert'; // __VERSION__ is injected by babel-plugin-version-inline
+
+// Returns `true` if the two versions are compatible
+/* global __VERSION__ */ export function validateLoaderVersion(loader) {
```
Comment (ibgreen) [inline modules/core/src/utils/version.js]: Nit: global at top

File: modules/loader-utils/src/worker-utils/create-worker.js [hunk 3]
```diff
@@ -88,7 +91,7 @@ async function parseData({loader, arraybuffer, byteOffset, byteLength, options,
 
 // Filter out noise messages sent to workers
 function isKnownMessage(data, name) {
-  return data && data.type === 'parse' && data.source === 'loaders.gl';
+  return data && data.type === 'parse' && data.source && data.source.startsWith('loaders.gl');
```
Comment (ibgreen) [inline modules/loader-utils/src/worker-utils/create-worker.js]: Nit: Do we even need to check for `parse` in this case?

File: modules/loader-utils/src/worker-utils/version.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
+// Returns `true` if the two versions are compatible
+export function validateLoaderVersion(coreVersion, loader) {
```
Comment (ibgreen) [inline modules/loader-utils/src/worker-utils/version.js]: Why do we need to validate both in core and loaders?

File: test/browser.js [hunk 5]
```diff
@@ -29,7 +29,7 @@ test.onFailure(window.browserTestDriver_fail);
 
 // This constant will be inlined by babel plugin.
 // To test source without transpilation, set a fallback here.
-window.__VERSION__ = 'latest';
+window.__VERSION__ = require('../lerna.json').verion;
```
Comment (ibgreen) [inline test/browser.js]: Typo: `version`

File: test/node.js [hunk 6]
```diff
@@ -12,7 +12,7 @@ if (typeof process !== 'undefined') {
 
 // This constant will be inlined by babel plugin.
 // To test source without transpilation, set a fallback here.
-global.__VERSION__ = 'latest';
+global.__VERSION__ = require('../lerna.json').verion;
```
Comment (ibgreen) [inline test/node.js]: Typo: `version`

File: modules/loader-utils/src/worker-utils/create-worker.js (hunk 3 above)
Comment (Pessimistress) [reply modules/loader-utils/src/worker-utils/create-worker.js]: Yes, the worker can also receive message from its previously submitted parse requests to sub workers.

File: modules/loader-utils/src/worker-utils/version.js (hunk 4 above)
Comment (Pessimistress) [reply modules/loader-utils/src/worker-utils/version.js]: The worker loaders do not use core.

**Pull request merged**